mvlearn/semi_supervised/ctclassifier.py

```diff
@@ -178,20 +178,20 @@ def __init__(self,
         super().__init__(estimator1, estimator2, random_state)
 
         # if not given, set classifiers as gaussian naive bayes estimators
-        if self.estimator1_ is None:
-            self.estimator1_ = GaussianNB()
-        if self.estimator2_ is None:
-            self.estimator2_ = GaussianNB()
+        if self.estimator1 is None:
+            self.estimator1 = GaussianNB()
+        if self.estimator2 is None:
+            self.estimator2 = GaussianNB()
 
         # If only 1 of p or n is not None, set them equal
         if (p is not None and n is None):
             n = p
-            self.p_, self.n_ = p, n
+            self.p, self.n = p, n
         elif (p is None and n is not None):
             p = n
-            self.p_, self.n_ = p, n
+            self.p, self.n = p, n
         else:
-            self.p_, self.n_ = p, n
+            self.p, self.n = p, n
```

Comment:
this should be done in the fit. There should be no logic in the init just do:

self.a = a
self.b = b

etc.